### ISSUE TITLE
Rebase workflow doesn't work when testing locally

### ISSUE THREAD

**Issue by jeroenvisser101** [danger/danger#163]
Since the local version works by checking for merge commits, it will not work when you use a workflow that involves rebasing to get things into the base branch.

However, it seems that there's a way to see to which branch a certain commit belonged, and we might be able to base danger's diff on that.

**Comment by ffittschen** [comment]
I'm just leaving these stacktraces here, because this is what got printed out, when I ran into this issue and this way the issue is easier to find 😉
```
$ bundle exec danger local
No recent pull requests found for this repo, danger requires at least one PR for the local mode. (RuntimeError)
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/ci_source/local_git_repo.rb:52:in `initialize'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/danger_core/environment_manager.rb:12:in `new'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/danger_core/environment_manager.rb:12:in `block in initialize'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/2.2.0/set.rb:283:in `each_key'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/2.2.0/set.rb:283:in `each'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/danger_core/environment_manager.rb:9:in `initialize'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/commands/local.rb:29:in `new'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/commands/local.rb:29:in `run'
```
```
$ LOCAL_GIT_PR_ID=1
$ bundle exec danger local
Could not find the pull request (1) inside the git history for this repo. (RuntimeError)
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/ci_source/local_git_repo.rb:50:in `initialize'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/danger_core/environment_manager.rb:12:in `new'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/danger_core/environment_manager.rb:12:in `block in initialize'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/2.2.0/set.rb:283:in `each_key'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/2.2.0/set.rb:283:in `each'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/danger_core/environment_manager.rb:9:in `initialize'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/commands/local.rb:29:in `new'
    /Users/florian/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/danger-0.8.3/lib/danger/commands/local.rb:29:in `run'
```

**Comment by orta** [comment]
For a rebased PR, are there any indications inside the git commit history for the start/end of a PR? ( I have no idea ) would it be represented by one commit?

**Comment by jeroenvisser101** [comment]
GitHub does seem to know, because git-sha appears in a PR, but I don't think you can without querying GitHub for this info.

**Comment by orta** [comment]
Perhaps we should is to pull that data from the PR instead of trying to derive it from the commit log then?

**Comment by jeroenvisser101** [comment]
Not sure if all our sources (gitlab, bitbucket, github) would have the same data there though

**Comment by orta** [comment]
Currently `danger local` supports only GitHub, ( because of the commit thing )  in theory, this gives other the chance to also work

**Comment by jeroenvisser101** [comment]
That data doesn't seem to be available through the API yet, that might be an issue

**Comment by orta** [comment]
Would be fixed by #382
